services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -454,6 +470,8 @@ public void setRule(Rule rule) {
     }
 
     public void removeRule(String ruleId) {
+        Set<Rule> newRules = allRules.stream().filter(rule -> !rule.getItemId().equals(ruleId)).collect(Collectors.toSet());
+        allRules = new ArrayList<>(newRules);
```

Comment:
Don't we risk concurrency issue in case of multiple update of this list ?
in case there is the job running in // or multiple threads using the list ?